pkg/plugins/generic/generic_plugin.go

```diff
@@ -368,76 +364,9 @@ func (p *GenericPlugin) needRebootNode(state *sriovnetworkv1.SriovNetworkNodeSta
 		log.Log.V(2).Info("generic plugin needRebootNode(): need reboot for updating kernel arguments")
 		needReboot = true
 	}
-
-	// Create a map with all the PFs we will need to configure
-	// we need to create it here before we access the host file system using the chroot function
-	// because the skipConfigVf needs the mstconfig package that exist only inside the sriov-config-daemon file system
-	pfsToSkip, err := getPfsToSkip(p.DesireState, p.helpers)
-	if err != nil {
-		return false, err
-	}
-	p.pfsToSkip = pfsToSkip
-
-	updateNode, err = p.helpers.WriteSwitchdevConfFile(state, p.pfsToSkip)
-	if err != nil {
-		log.Log.Error(err, "generic plugin needRebootNode(): fail to write switchdev device config file")
-		return false, err
-	}
-	if updateNode {
-		log.Log.V(2).Info("generic plugin needRebootNode(): need reboot for updating switchdev device configuration")
-		needReboot = true
-	}
-
 	return needReboot, nil
 }
 
-// getPfsToSkip return a map of devices pci addresses to should be configured via systemd instead if the legacy mode
-// we skip devices in switchdev mode and Bluefield card in ConnectX mode
-func getPfsToSkip(ns *sriovnetworkv1.SriovNetworkNodeState, mlxHelper mlx.MellanoxInterface) (map[string]bool, error) {
-	pfsToSkip := map[string]bool{}
-	for _, ifaceStatus := range ns.Status.Interfaces {
-		for _, iface := range ns.Spec.Interfaces {
-			if iface.PciAddress == ifaceStatus.PciAddress {
-				skip, err := skipConfigVf(iface, ifaceStatus, mlxHelper)
-				if err != nil {
-					log.Log.Error(err, "GetPfsToSkip(): fail to check for skip VFs", "device", iface.PciAddress)
-					return pfsToSkip, err
-				}
-				pfsToSkip[iface.PciAddress] = skip
-				break
-			}
-		}
-	}
-
-	return pfsToSkip, nil
-}
-
-// skipConfigVf Use systemd service to configure switchdev mode or BF-2 NICs in OpenShift
-func skipConfigVf(ifSpec sriovnetworkv1.Interface, ifStatus sriovnetworkv1.InterfaceExt, mlxHelper mlx.MellanoxInterface) (bool, error) {
```

Comment:
how we handle the BF cards if we remove this one?

Reply:
I assume there is no need for special hanling of BF cards anymore. Creation of VFs should work in both BF modes(DPU and NIC) when the operator is in any configuration mode(in daemon or in systemd mode). If you need to create VFs on boot, e.g. for hw-offloading use-case, you need to switch the operator to systemd mode.

Did I miss something?

Reply:
we are running some validations internally base on OCP deployment and BF2 just to be sure everything continue to work as expected 

cc @zeeke